File: SAMSUI/src/app/Model/master/designation.ts
import { BaseModel } from '../base/baseModel';

export class DesignationModel extends BaseModel{

    public designationId ?:number;

    public locationId ?:number;

    public designationName ?:string;

    public designationDesc ?:string;

    public active ?:string;

    public activeDisp ?:boolean;

    public createdBy ?: string;

    public createdDt ?: Date;

    public createdDtDisp ?: string;
    
    public updatedBy ?: string;

    public updatedDt ?: Date;
    
    public updatedDtDisp ?: string;
    
    public pageNumber ?: number;
    
    public recordsPerPage ?: number;

    public isListEmpty ?:boolean;
    
}

File: sams-backend/src/main/java/com/sams/entity/Designation.java
package com.sams.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "m_designation", schema = "base")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Designation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "designation_id")
    private Long designationId;

    @Column(name = "location_id")
    private Long locationId;

    @Column(name = "designation_name")
    private String designationName;

    @Column(name = "designation_desc")
    private String designationDesc;

    @Column(name = "active")
    private String active;

    @Column(name = "active_disp")
    private Boolean activeDisp;

    @Column(name = "created_by")
    private String createdBy;

    @Column(name = "created_dt")
    private LocalDateTime createdDt;

    @Column(name = "created_dt_disp")
    private String createdDtDisp;

    @Column(name = "updated_by")
    private String updatedBy;

    @Column(name = "updated_dt")
    private LocalDateTime updatedDt;

    @Column(name = "updated_dt_disp")
    private String updatedDtDisp;

    @PrePersist
    protected void onCreate() {
       // Auto-generated
    }

    @PreUpdate
    protected void onUpdate() {
       // Auto-generated
    }
}